### PR TITLE
BUGFIX: Search by defined property is not possible

### PR DESCRIPTION
**Pull request by lars85**
- `findByProperties` should not search for a node identifier.
- `$term` in `foreach ($searchTerm as $term) {` has overwritten `$term` from methode signature. Because of that it was not possible to pass an array to `$this->nodeDataRepository->findByProperties`

### CODE REVIEW

**Review by markusguenther — COMMENTED** [review]
Thanks @lars85 for this PR. It looks good by reading, but would be nice if some one else can verify that it is not breaking something. But as it is not used somewhere it looks like kind of dead code.

File: Neos.Neos/Classes/Domain/Service/NodeSearchService.php [hunk 1]
```diff
@@ -49,32 +48,17 @@ class NodeSearchService implements NodeSearchServiceInterface
...
      *
-     * @param string|array $term search term
+     * @param string|array $term search '%term%' (string) OR '%"property": "value"%' (array) in properties
      * @param array $searchNodeTypes
      * @param Context $context
      * @param NodeInterface $startingPoint
      * @return array <\Neos\ContentRepository\Domain\Model\NodeInterface>
      */
-    public function findByProperties($term, array $searchNodeTypes, Context $context, NodeInterface $startingPoint = null)
+    public function findByProperties($term, array $searchNodeTypes, Context $context, NodeInterface $startingPoint = null): array
```
Comment (daniellienert) [inline Neos.Neos/Classes/Domain/Service/NodeSearchService.php:57]: That does not match the current `NodeSearchServiceInterface`

**Review by kdambekalns — CHANGES_REQUESTED** [review]
Good find with the overridden `$term`, but I'd like to see this solved differently.

- keep the exception - or do an early return - because the method should not be used to find "everything" (which is what I'd expect for an empty searach term)
- keep the node identifier search! "does not seem to be used" is a bit lax for justification
- pass `$searchTerm` (not `$term`) to the `NodeDataRepository` to fix the actual bug

**Comment by kdambekalns** [comment]
Closing in favor of #3429

**Pull request closed without merging**